src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -961,6 +961,77 @@ public static async ValueTask<FormulaValue> AsType(EvalVisitor runner, EvalVisit
             {
                 return new ErrorValue(irContext, e.ExpressionError);
             }
+        }
+
+        public static async ValueTask<FormulaValue> SearchImpl(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
+        {
+            var source = args[0];
+            var textToSearchArg = args[1];
+
+            if (source is BlankValue || source is ErrorValue || textToSearchArg is BlankValue)
+            {
+                return source;
+            }
+
+            var sourceTable = (TableValue)args[0];
+            var textToSearch = ((StringValue)textToSearchArg).Value.ToLower(runner.CultureInfo);

```

Comment:
>((StringValue)textToSearchArg) [](http://example.com/codeflow?start=31&length=30)

what if textToSearch is error?  (did IR pipeline already check that)

Reply:
The StandardErrorHandlingAsync wrapper has already blocked this, we shouldn't get error arguments here.